### PR TITLE
remove portMapEquals in service

### PR DESCRIPTION
**Pull request by yeedove**
**Please provide a description of this PR:**

portMapEquals can be replaced by maps.Equals.

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -366,24 +366,12 @@ func WorkloadInstancesEqual(first, second *WorkloadInstance) bool {
...
 		return false
 	}
-	if !portMapEquals(first.PortMap, second.PortMap) {
+	if !maps.Equal(first.PortMap, second.PortMap) {
 		return false
 	}
 	return true
 }
 
-func portMapEquals(a, b map[string]uint32) bool {
```
Comment (wulianglongrd) [inline pilot/pkg/model/service.go:375]: `maps.Equal` checks whether the key exists, but `portMapEquals` does not. This causes the following examples to have different results.

```code
a := map[string]uint32{"a": 0}
b := map[string]uint32{"b": 0}

portMapEquals(a, b) // got: true
maps.Equal(a, b)    // got: false
````

Comment (yeedove) [reply pilot/pkg/model/service.go:375]: I don't think there is a case where the value of the PortMap, which can also be described as the port, should be 0.

Comment (yeedove) [reply pilot/pkg/model/service.go:375]: I don't think there is a case where the value of the PortMap, which can also be described as the port, should be 0.

Comment (hzxuzhonghu) [reply pilot/pkg/model/service.go:375]: You can think that `portMapEquals` is not right

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**